### PR TITLE
DC Bot Token Info and DC Bot Token deleter

### PR DESCRIPTION
**Pull request by Tert0**
**Description**
- A Discord Bot Token info
- A Disocrd Bot Token deleter

**Additional Notes**
Does anyone have a better idea for the info text?

### CODE REVIEW

**Comment by luwol03** [comment]
Could you send the user the deleted message via PN so that he does not have to rewrite it? Maybe you can even replace the token with `***`.

**Comment by luwol03** [comment]
Your regex does not work in all cases.
```
"ODDFOFXUpgf7yEntul5ockCA.OFk6Ph.lmsA54bT0Fux1IpsYvey5XuZk04"
``` 
*(its a fake token)*
this won't be deleted.

**Comment by Tert0** [comment]
I have already found and fixed the bug, but have not pushed it yet.

**Review by florian-sabonchi — DISMISSED** [review]

File: morpheushelper/cogs/discord_bot_tokens.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+
+def make_embed(requested_by: Member) -> Embed:
+    embed = Embed(title=translations.discordbottoken_title, colour=Colours.DiscordBotToken)
+    embed.description = translations.discordbottoken_description
+    embed.set_footer(text=translations.f_requested_by(requested_by, requested_by.id), icon_url=requested_by.avatar_url)
+    return embed
+
+
+class DiscordBotTokenCog(Cog):
```
Comment (ghost) [inline morpheushelper/cogs/discord_bot_tokens.py:20]: Please consider to create a class var with a compiled version of the regex, like so:
```python
class DiscordBotTokenCog(Cog):
    RE_DC_TOKEN = re.compile(r'[A-Za-z\d]{24}\.[A-Za-z\d]{6}\.[A-Za-z\d\-\_]{27}')
```

File: morpheushelper/cogs/discord_bot_tokens.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        except (Forbidden, NotFound, HTTPException):
+            pass
+
+    @guild_only()
+    async def on_message(self, msg: Message):
+        """
+        deletes a message if it contains a discord bot token
+        """
+
+        if re.match(r'[A-Za-z\d]{24}\.[A-Za-z\d]{6}\.[A-Za-z\d\-\_]{27}', msg.content):
```
Comment (ghost) [inline morpheushelper/cogs/discord_bot_tokens.py]: An early return would be nice an save on nested for the rest of the method.
```python
if not self.RE_DC_TOKEN.match(msg.content):
    return

# ...
```

File: morpheushelper/cogs/discord_bot_tokens.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    async def token(self, ctx: Context):
+        """
+        display information about discord bot toktens
+        """
+
+        await ctx.send(embed=make_embed(ctx.author))
+        try:
+            await ctx.message.delete()
+        except (Forbidden, NotFound, HTTPException):
+            pass
```
Comment (ghost) [inline morpheushelper/cogs/discord_bot_tokens.py]: logging would be nice.

File: morpheushelper/cogs/discord_bot_tokens.py [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+        """
+        deletes a message if it contains a discord bot token
+        """
+
+        if re.match(r'[A-Za-z\d]{24}\.[A-Za-z\d]{6}\.[A-Za-z\d\-\_]{27}', msg.content):
+            await msg.channel.send(embed=make_embed(msg.author))
+            try:
+                await msg.delete()
+            except (Forbidden, NotFound, HTTPException):
+                pass
```
Comment (ghost) [inline morpheushelper/cogs/discord_bot_tokens.py]: logging would be nice.

**Review by ghost — CHANGES_REQUESTED** [review]

File: morpheushelper/cogs/discord_bot_tokens.py [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+        except (Forbidden, NotFound, HTTPException) as error:
+            discord_token_logger.error(error)
+
+    @guild_only()
+    async def on_message(self, msg: Message):
+        """
+        deletes a message if it contains a discord bot token
+        """
+
+        if tokens := self.RE_DC_TOKEN.findall(msg.content):
```
Comment (ghost) [inline morpheushelper/cogs/discord_bot_tokens.py]: An early return would be nice an save on nested for the rest of the method.
```python
if not (tokens := self.RE_DC_TOKEN.findall(msg.content)):
    return

await msg.channel.send(embed=make_embed(msg.author))
# ...
```

**Review by ghost — CHANGES_REQUESTED** [review]

**Review by ghost — APPROVED** [review]

File: morpheushelper/cogs/discord_bot_tokens.py [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+        display information about discord bot toktens
+        """
+
+        await ctx.send(embed=make_embed(ctx.author))
+        try:
+            await ctx.message.delete()
+        except (Forbidden, NotFound, HTTPException) as error:
+            discord_token_logger.error(error)
+
+    @guild_only()
```
Comment (Defelo) [inline morpheushelper/cogs/discord_bot_tokens.py]: you can't use this decorator on event handlers

File: morpheushelper/cogs/discord_bot_tokens.py [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+    async def on_message(self, msg: Message):
+        """
+        deletes a message if it contains a discord bot token
+        """
+
+        if self.RE_DC_TOKEN.findall(msg.content):
+            return
+        await msg.author.send(embed=make_embed(msg.author))
```
Comment (Defelo) [inline morpheushelper/cogs/discord_bot_tokens.py]: when deleting a message you should send a different text to explain to the user why the message has been deleted

File: translations/en.yml [hunk 8]
```diff
@@ -771,3 +771,7 @@ run_usage: |
   your code
   `​`​`
 run_output: Run Output
+discordbottoken_title: Discord Bot Token
+discordbottoken_description: >
+  Discord Bot Tokens are used for authentication.
+  If another person gets it, the entire bot can be controlled with it!
```
Comment (Defelo) [inline translations/en.yml]: please improve this text and add information on how to reset the bot token (include a link to the developer portal!)

**Review by Defelo — CHANGES_REQUESTED** [review]

**Review by ghost — APPROVED** [review]

**Review by ghost — CHANGES_REQUESTED** [review]
Due to a the restructuring of the Bot this pull reuest is not up to date. Feel free to close this and reopen it in the [cogs-repo](https://github.com/PyDrocsid/cogs).

**Pull request closed without merging**